File: src/main/java/domain/Song.java
package domain;

import java.util.List;

public class Song {

    private int songId;
    private Author author;
    private int releaseYear;
    private Rate rate;
    private List<Comment> comments;
    private String musicGenre;

    public Author getAuthor() {
        return author;
    }

    public void setAuthor(Author author) {
        this.author = author;
    }

    public int getReleaseYear() {
        return releaseYear;
    }

    public void setReleaseYear(int releaseYear) {
        this.releaseYear = releaseYear;
    }

    public Rate getRate() {
        return rate;
    }

    public void setRate(Rate rate) {
        this.rate = rate;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public void setComments(List<Comment> comments) {
        this.comments = comments;
    }

    public String getMusicGenre() {
        return musicGenre;
    }

    public void setMusicGenre(String musicGenre) {
        this.musicGenre = musicGenre;
    }
}

File: frontend/src/app/song/song.ts
export class Song{

  songId:number;
  songName: string;
  musicGenre:string;
  releaseYear: number;
  youTubeLink:string;
  authorId:number;
  songAverageRate:number;


  constructor(songId: number, songName: string, musicGenre: string, releaseYear: number, youTubeLink: string, authorId: number, songAverageRate:number){
    this.songId = songId;
    this.songName = songName;
    this.musicGenre = musicGenre;
    this.releaseYear = releaseYear;
    this.youTubeLink = youTubeLink;
    this.authorId=authorId;
    this.songAverageRate=songAverageRate;
  }
}
